src/store/reducer/cartControlSlice.ts

```diff
@@ -0,0 +1,28 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface CartControlState {
+  isCartOpen: boolean;
+}
+
+const initialState: CartControlState = {
+  isCartOpen: false,
+};
+
+export const CartControlSlice = createSlice({
+  name: "CartControl",
+  initialState,
+  reducers: {
+    switchCartDisplay: (state: CartControlState) => {
+      state.isCartOpen = !state.isCartOpen;
+    },
+    setCartDisplayState: (state: CartControlState, action: PayloadAction<boolean>) => {
+      state.isCartOpen = action.payload;
+    },
+  },
+});
+
+export const { switchCartDisplay, setCartDisplayState } = CartControlSlice.actions;
+
+export const getCartDisplayCondition = (state: CartControlState) => state.isCartOpen;
+
+export default CartControlSlice.reducer;
```

Comment:
Is there any reason why create another identical file almost the same as ShoppingCartSlice?

Reply:
OMG...Sorry we did not check our code carefully before the PR. The shoppingCartSlice.ts can be removed.

Reply:
> Is there any reason why create another identical file almost the same as ShoppingCartSlice?

OMG...Sorry we did not check our code carefully before the PR. The shoppingCartSlice.ts can be removed.

Reply:
> Is there any reason why create another identical file almost the same as ShoppingCartSlice?

ShoppingCartSlice.ts was deleted.